otherlibs/stdune/src/copyfile_stubs.c

```diff
@@ -85,12 +89,43 @@ static int dune_sendfile(int in, int out, size_t length) {
   return length;
 }
 
+static ssize_t dune_copy_file_range(int in, int out, size_t length) {
+  ssize_t ret;
+  while (length > 0) {
+    ret = copy_file_range(in, NULL, out, NULL, length, 0);
+    if (ret < 0) {
+      return ret;
+    }
+    length = length - ret;
+  }
+  return length;
+}
+
+static int kernel_version(void) {
+  struct utsname uts;
+  int major, minor, patch;
+
+  if (uname(&uts) < 0)
+    return -1;
+
+  if (sscanf(uts.release, "%d.%d.%d", &major, &minor, &patch) != 3)
+    return -1;
+
+  return KERNEL_VERSION(major, minor, patch);
+}
+
 CAMLprim value stdune_sendfile(value v_in, value v_out, value v_size) {
+  static ssize_t (*dune_copyfile)(int, int, size_t) = NULL;
+  if (dune_copyfile == NULL) {
+    if (kernel_version() < KERNEL_VERSION(5, 19, 0)) {
+      dune_copyfile = &dune_sendfile;
```

Comment:
My understanding was that there was no reason to not use `FICLONE` starting from kernel version `4.5`. I don't know how relevant these kinds of systems are but if it is easy enough to add it would be great to include that case.